paimon-core/src/main/java/org/apache/paimon/iceberg/IcebergOptions.java

```diff
@@ -48,11 +48,22 @@ public class IcebergOptions {
                     .defaultValue(50);
 
     /** Where to store Iceberg metadata. */
-    public enum StorageType {
-        // Store Iceberg metadata with each table.
-        PER_TABLE,
-        // Store Iceberg metadata in a separate directory.
-        // This directory can be specified as the Iceberg warehouse directory.
-        ICEBERG_WAREHOUSE
+    public enum StorageType implements DescribedEnum {
+        DISABLED("Disable Iceberg compatibility support."),
+        TABLE_LOCATION("Store Iceberg metadata with each table."),
+        HADOOP_CATALOG(
+                "Store Iceberg metadata in a separate directory. "
+                        + "This directory can be specified as the warehouse directory of an Iceberg Hadoop catalog.");
+
```

Comment:
You can add a value and implement `toString`. For the paimon style, the value should be `table-location` and `hadoop-catalog`.